### ISSUE TITLE
Multiple errors during execution of large Hunt-Jobs

### ISSUE THREAD

**Issue by mammo0** [aol/moloch#1377]
**Elasticsearch version**:
7.5.1

**Moloch version**:
2.2.1

**OS name and version**:
Ubuntu 18.04
in Docker container (see: https://github.com/mammo0/docker-moloch)

**How was Moloch built/installed:** (rpm, deb, easybutton, ...)
deb

**Provide logs, stack traces and steps to reproduce**:

Hello,

in my Moloch environment I've got a large amount of network data to search through with a hunt job (~10.000.000 sessions). At first the job ran fine, but at around 30% it threw the following error:

```
Hunt error searching sessions: [search_context_missing_exception] No search context found for id [52501] (and) [...]
```

After a little search I found out that this kind of error is thrown by Elasticsearch if a timeout is hit... But I don't figured out what timeout exactly, and where it can be increased. Maybe you've got an idea.

After that I tried to simply rerun/continue the above hunt job. It immediately threw another error:
```
 Hunt error searching sessions: [illegal_argument_exception] Batch size is too large, size must be less than or equal to: [10000] but was [6222549]. Scroll batch sizes cost as much memory as result windows so they are controlled by the [index.max_result_window] index level setting. (and) [...]
```

I figured out that the above number `6222549` was calculated in the line
https://github.com/aol/moloch/blob/b2d8e312a89f658741c29c79b8f035d52c344859/viewer/viewer.js#L7260
I don't know why the query size is calculated here... If I simply comment out this line (resp. including the surrounding `if`-block), the job started normally again. Now the job is running and I will report later if another error occurs.

But until then you may have some ideas regarding the first error above and can explain why this query size calculation is done.

Thank you.

**Comment by mammo0** [comment]
> Now the job is running and I will report later if another error occurs.

Immediately after I submitted this issue, the restarted process failed again with the same error message as the first time:
```
Hunt error searching sessions: [search_context_missing_exception] No search context found for id [759] (and) [...]
```

I saw that in viewer.js there is the value `esQueryTimeout`:
https://github.com/aol/moloch/blob/b2d8e312a89f658741c29c79b8f035d52c344859/viewer/viewer.js#L70

By default it is set to 300s resp. 5min. I increased the value in the config.ini file. The option `elasticsearchTimeout` didn't exist in the `[default]` section of the config.ini, so I added it with a value of 900 (15min).
I will report back if this helps.

**Comment by mammo0** [comment]
Ok, the option `elasticsearchTimeout` didn't solve my problem. So I removed it again from my config.ini.

Meanwhile I found a topic in the Elasticsearch docs: https://www.elastic.co/guide/en/elasticsearch/reference/7.5/search-request-body.html#scroll-search-context

I looked into the viewer.js code and saw that the `scroll` parameter is hard coded to 600s. Is this intended and if so, why? Maybe this can be set by an option in config.ini?

I will test if it works if I increase this value and report back.

**Comment by mammo0** [comment]
Since yesterday I get another error message:

```
Error hunting on remote viewer: Error: socket hang up
```

I found out this happens because the viewer.js function `isLocalView(...)` concludes that it's not the local view and therefore it runs the `noCb` callback function. This means a remote request is called (via the `makeRequest(...)` function). And this standard NodeJS request sometimes runs into the default timeout of [2 minutes](https://nodejs.org/docs/latest-v10.x/api/http.html#http_server_timeout).

The `isLocalView` function assumes that I'm on a remote View because I use Moloch in a Docker environment. And every time I upgrade the container, it gets a new random hostname.
I will try to start the viewer in the docker container with `-h localhost` to override the default hostname. Hope this helps.

**Comment by mammo0** [comment]
Now it seems that the hunt jobs run as expected.

If they are all finished I will open a pull request.

**Comment by mammo0** [comment]
Issue fixed during discussion in PR #1385.
Thank you!
